src/xcode/ENA/ENA/Source/Services/HealthCertificate/HealthCertificateService.swift

```diff
@@ -493,9 +499,26 @@ class HealthCertificateService {
 						}
 					}
 				}
+				
+				self.updateNotifications()
 			}
 			.store(in: &subscriptions)
 	}
+	
+	/// This method should be called: At startup, at creation, at removal and at update validity states of HealthCertificates.
+	/// First, removes all local notifications and then re-adds all updates or new notifications to the notification center.
+	func updateNotifications() {
```

Comment:
I only see the call for the case of updating validity states.
Is there a task to call this func in the other cases? So we don't forget to wire this up?

Reply:
The update shall be triggered when the app config changes or the validity changes or at startup. In every of these cases, `updateValidityStatesAndNotifications` was already called so i added the `updateNotifications` just at the end of the `updateValidityStatesAndNotifications` function to cover these three scenarios.